buildSrc/src/main/java/com/google/firebase/gradle/plugins/UpdatePinnedDependenciesTask.kt

```diff
@@ -0,0 +1,115 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.gradle.plugins
+
+import java.io.File
+import org.gradle.api.DefaultTask
+import org.gradle.api.provider.Property
+import org.gradle.api.tasks.InputFile
+import org.gradle.api.tasks.OutputFile
+import org.gradle.api.tasks.StopExecutionException
+import org.gradle.api.tasks.TaskAction
+
+/**
+ * Changes the project level dependencies in a project's build file to pinned dependencies.
+ *
+ * Computes project level dependencies outside of the project's library group, looks for instances
+ * of them in the `dependencies` block in the build file, and replaces them with pinned versions.
+ * GMaven is used to figure out the latest version to use for a given dependency.
+ *
+ * For example, given the following input:
+ * ```
+ * dependencies {
+ *    implementation(project(":appcheck:firebase-appcheck-interop"))
+ * }
+ * ```
+ *
+ * The output would be:
+ * ```
+ * dependencies {
+ *    implementation("com.google.firebase:firebase-appcheck-interop:17.0.1")
+ * }
+ * ```
+ *
+ * *Assuming that `17.0.1` is the latest version of `firebase-appcheck-interop`*
+ *
+ * @property buildFile A [File] that should be used as the source to update from. Typically the
+ * `build.gradle` or `build.gradle.kts` file for a given project.
+ * @property outputFile A [File] that should be used to write the new text to. Typically the same as
+ * the input file ([buildFile]).
+ * @see PostReleasePlugin
+ */
+abstract class UpdatePinnedDependenciesTask : DefaultTask() {
+  @get:[InputFile]
+  abstract val buildFile: Property<File>
+
+  @get:[OutputFile]
+  abstract val outputFile: Property<File>
+
+  @TaskAction
+  fun build() {
+    val needsChanging = computeLibrariesThatNeedChanging()
```

Comment:
what you are computing is the "dependencies" that need to be updated, right?

If so, use a naming that closer reflects that, like "getOutdatedProjectLevelDependencies"